src/lib/profiles/data-management/Current/SubscriptionEngine.cpp

```diff
@@ -283,6 +283,72 @@ WEAVE_ERROR SubscriptionEngine::NewClient(SubscriptionClient ** const appClient,
     return NewClient(appClient, apBinding, apAppState, aEventCallback, apCatalog, aInactivityTimeoutDuringSubscribingMsec, NULL);
 }
 
+#if WEAVE_CONFIG_PERSIST_SUBSCRIPTION_STATE
+WEAVE_ERROR SubscriptionEngine::NewClientFromPersistedState(SubscriptionClient ** const appClient, Binding * const apBinding, void * const apAppState,
+                                                            SubscriptionClient::EventCallback const aEventCallback,
+                                                            const TraitCatalogBase<TraitDataSink> * const apCatalog,
+                                                            const uint32_t aInactivityTimeoutDuringSubscribingMsec, TLVReader & reader)
+{
+    WEAVE_ERROR err = WEAVE_ERROR_NO_MEMORY;
+
```

Comment:
The passed-in Binding object needs to be checked to be not NULL. It is deferenced in LoadFromPersistedState.

Reply:
Resolving this conversation with the note that we need a NULL check.

Reply:
Sorry for the delay, just updated~

Reply:
sorry I just realized here this binding is a return pointer, this is passed in as null and returned to application.

Reply:
returned to application because it feel like subscription handler and client can share the same binding.

Reply:
sorry that was my brain fart. it was allocated and returned here https://nest-internal-review.git.corp.google.com/c/platform/nlweavetools/+/206819/4/src/wdm/nlwtwdmsubscriber.cpp
just put the null check back